### PR TITLE
Add removeTransaction implementation with test coverage

### PR DESCRIPTION
**Pull request by MahmoodTarek**
**Implement and test removeTransaction functionality**

1.  Added implementation for removeTransaction in InMemoryTransactionDataSourceImpl
2.  Created TransactionRemoveTest.kt to validate different test cases for transaction removal
3.  Ensured edge cases are handled: non-existent transaction, empty list, etc.
4.   Used 'check' function for readable test output using "when...then..." format

### CODE REVIEW

File: Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt [hunk 1]
```diff
@@ -4,14 +4,15 @@ import datasource.TransactionDataSource
...
+    val transactionList = mutableListOf<Transaction>()
+
     override fun createTransaction(transaction: Transaction) {
         TODO("Not yet implemented")
     }
 
-    override fun removeTransaction(transaction: Transaction) {
-        TODO("Not yet implemented")
-    }
+    override fun removeTransaction(transaction: Transaction) =
```
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt:14]: it's better to use transactionList.remove(transaction) and it return true if removing success or false if failed

File: Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt
Comment (AbdElrahman-Rafaat-Amer) [inline Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt:1]: Here u need to provide the items u want to delete to see if they really match ur test cases.

File: Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt (hunk 1 above)
Comment (MahmoodTarek) [reply Personal-Finance-Tracker/src/main/kotlin/datasource/InMemoryTransactionDataSourceImpl.kt:14]: Thanks bro! 
You're right that `transactionList.remove(transaction)` works,  
but in our case, I used `removeIf { it.id == transaction.id }` because it only checks the `id`, not the whole object.

Since `Transaction` is a data class, `remove(transaction)` compares all fields (name, amount, date...),  
so if I change just one field (like the name), it won't match and won't remove it.

But `removeIf` will still work as long as the ID is the same — and in our app, ID is what really matters for identifying transactions.

Appreciate the feedback though!

**Review by DinaOthman21 — APPROVED** [review]

**Review by Hossam-Haroon — APPROVED** [review]

**Review by ALiyASSER0 — APPROVED** [review]
good job

File: Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        expected =  false
+    )
+    check(
+        name = "When removing one of multiple transactions, then should return true",
+        result= true,
+        expected = true
+    )
+}
+
+fun <T>check(name: String, result: T, expected: T) {
```
Comment (Mohamed-Elhanafy) [inline Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt]: we can use the check function that exist in the utils

**Review by Mohamed-Elhanafy — APPROVED** [review]

**Review by Mohamed-Elhanafy — APPROVED** [review]

File: Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt (hunk 2 above)
Comment (karim91r) [reply Personal-Finance-Tracker/src/main/kotlin/test/TransactionRemoveTest.kt]: done
